### PR TITLE
Backport patch for C++20 compatibility

### PR DESCRIPTION
**Pull request by mering**
This backports https://github.com/grpc/grpc/pull/36016 to latest version of gRPC available in BCR.

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (grpc) have been updated in this PR. Please review the changes.

File: modules/grpc/1.62.1.bcr.1/source.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "integrity": "sha256-yfmubk1vQEZO6ZWL5AaAh4ge1qo34w0OZNQO17453QE=",
+    "strip_prefix": "grpc-1.62.1",
+    "url": "https://github.com/grpc/grpc/archive/refs/tags/v1.62.1.tar.gz",
+    "patch_strip": 1,
+    "patches": {
+        "add_module_bazel.patch": "sha256-QA+8LXhdL6PhmaaQVLj9YN8+I02Y3gtgrUJghnhbEpc=",
```
Comment (fmeum) [inline modules/grpc/1.62.1.bcr.1/source.json:7]: Could you replace the unmodified patches with relative paths a la `../../1.62.1/patches/...`? That will make the diff much smaller.

Comment (mering) [reply modules/grpc/1.62.1.bcr.1/source.json:7]: How about the version change? Should I add an additional patch to change the version in `MODULE.bazel` then?

Comment (fmeum) [reply modules/grpc/1.62.1.bcr.1/source.json:7]: Ah, forgot about that. Your call, but please use this trick for the large other patch.

Comment (mering) [reply modules/grpc/1.62.1.bcr.1/source.json]: I added the extra patch and fixed the `update_integrity` tool along the way in https://github.com/bazelbuild/bazel-central-registry/pull/2163.

**Pull request merged**